package/kedro_viz/integrations/kedro/sqlite_store.py

```diff
@@ -60,10 +85,109 @@ def to_json(self) -> str:
         return json.dumps(session_dict)
 
     def save(self):
-        """Save the session store info on db ."""
+        """Save the session store info on db and uploads it to the cloud if a remote cloud path is provided ."""
         engine, session_class = create_db_engine(self.location)
         Base.metadata.create_all(bind=engine)
         database = next(get_db(session_class))
-        session_store_data = RunModel(id=self._session_id, blob=self.to_json())
+
+        session_store_data = RunModel(id=self._session_id, blob=self._to_json())
         database.add(session_store_data)
         database.commit()
+        if self.remote_location:
+            self._upload()
+
+    def _upload(self):
+        """Uploads the session store database file to the specified remote path on the cloud storage."""
+        db_name = _get_dbname()
+        try:
+            self._remote_fs.put(self.location, f"{self.remote_location}/{db_name}")
+        except Exception as exc:
+            logger.exception(exc)
+
+    def _download(self):
+        """Downloads all the session store database files from the specified remote path on the cloud storage
+        to your local project.
+        Note: All the database files are deleted after they are merged to the main session_store.db.
+        """
+        try:
+            # In theory we should be able to do this as a single operation:
+            # self._remote_fs.get(f"{self.remote_location}/*.db", str(Path(self.location).parent))
+            # but this does not seem to work correctly - maybe a bug in fsspec. So instead
+            # we do it in two steps.
+            remote_dbs = self._remote_fs.glob(f"{self.remote_location}/*.db")
+            self._remote_fs.get(remote_dbs, str(Path(self.location).parent))
+        except Exception as exc:
+            logger.exception(exc)
+
+    def _merge(self):
+        """Merges all the session store databases stored at the specified locations into the user's local session_store.db
+
+        Notes:
+        - This method uses multiple SQLAlchemy engines to connect to the user's session_store.db and to all the other downloaded dbs.
+        - It is assumed that all the databases share the same schema.
+        - In the version 1.0 - we only merge the runs table which contains all the experiments.
+        - The downloaded database files are deleted after it's runs are merged with the user's local session_store.db
+
+        Args:
+            database_location:  A list of local filepath in string format for all the databases
+
+        """
+        # Connect to the user's local session_store.db
+        engine, session_class = create_db_engine(self.location)
+        Base.metadata.create_all(bind=engine)
+        database = next(get_db(session_class))
+
+        all_runs_data = []
+        existing_run_ids = [id[0] for id in database.query(RunModel.id).all()]
```

Comment:
I think there must be some way to do this without the list comprehension.

Reply:
done